src/problems/MultipleSource/algo/matrix_bfs/intersection.py

```diff
@@ -106,7 +107,17 @@ def create_diag_matrices(self) -> Dict[str, Matrix]:
 
         return diag_matrices
 
-    def intersect_bfs(self) -> EpsilonNFA:
+    def create_masks_matrix(self) -> Matrix:
+        num_vert_graph = self.graph.get_number_of_vertices()
+        num_vert_regex = self.regular_automaton.num_states
+        num_verts_diag = num_vert_graph + num_vert_regex
+
+        mask_matrix = Matrix.identity(BOOL, num_vert_regex, value=True)
+        mask_matrix.resize(num_vert_regex, num_verts_diag)
+
+        return mask_matrix
+
+    def intersect_bfs(self, src_verts) -> EpsilonNFA:
         """
         Intersection implementation with synchronous breadth first traversal
```

Comment:
Looks like this function solves reachability problem, not automata intersection. Isn't it?